client/src/main/java/com/linecorp/decaton/client/DecatonClientImpl.java

```diff
@@ -91,4 +104,29 @@ public DecatonClientImpl(String topic,
     public void close() throws Exception {
         producer.close();
     }
+
+    private TaskMetadataProto convertToTaskMetadataProto(TaskMetaData overrideTaskMetadata) {
+        final TaskMetadataProto.Builder
+                taskMetadataProtoBuilder = TaskMetadataProto.newBuilder()
+                                                            .setSourceApplicationId(applicationId)
+                                                            .setSourceInstanceId(instanceId);
+        if (timestampSupplier != null) {
+            taskMetadataProtoBuilder.setTimestampMillis(timestampSupplier.get());
+        } else {
+            taskMetadataProtoBuilder.setTimestampMillis(System.currentTimeMillis());
+        }
+        if (overrideTaskMetadata == null) {
+            return taskMetadataProtoBuilder.build();
+        }
+        final Long timestamp = overrideTaskMetadata.getTimestamp();
+        final Long scheduledTime = overrideTaskMetadata.getScheduledTime();
+        if (timestamp != null && timestamp > 0) {
```

Comment:
If we juedge negative timestamp to be invalid here, I think we should throw `IllegalArgumentException` for it rather than silently dropping the value.
But I think we can leave the value of it as the responsibility of users, so we may just check if it's null or not here.